File: packages/flutter/lib/src/resources/guardrails.dart
/// Guardrails resource — `/api/guardrails`.
library;

import '../models.dart';
import 'base.dart';

class GuardrailsResource extends BaseResource {
  const GuardrailsResource(super.client);

  Future<List<Guardrail>> list({String? tag}) async {
    final data = await client.request(
      'GET',
      'guardrails',
      queryParameters: tag != null ? {'tag': tag} : null,
    );
    if (data is List) {
      return data.whereType<Map<String, dynamic>>().map(Guardrail.fromJson).toList();
    }
    return const [];
  }

  Future<Guardrail> get(String id) async {
    final data = await client.request('GET', 'guardrails/$id');
    return Guardrail.fromJson(data as Map<String, dynamic>);
  }

  Future<Guardrail> create(GuardrailCreate payload) async {
    final data = await client.request('POST', 'guardrails', body: payload.toJson());
    return Guardrail.fromJson(data as Map<String, dynamic>);
  }

  Future<Guardrail> update(String id, GuardrailUpdate payload) async {
    final data = await client.request('PUT', 'guardrails/$id', body: payload.toJson());
    return Guardrail.fromJson(data as Map<String, dynamic>);
  }

  Future<void> delete(String id) => client.request('DELETE', 'guardrails/$id');
}

File: packages/python/src/tbd_agents/resources/guardrails.py
from __future__ import annotations

from typing import Any, Mapping

from tbd_agents.types import Guardrail, GuardrailCreate, GuardrailUpdate

from .base import BaseResource


class GuardrailsResource(BaseResource):
    def create(self, payload: GuardrailCreate | Mapping[str, Any]) -> Guardrail:
        return self._post("/guardrails", payload, model=Guardrail)

    def list(self, *, tag: str | None = None) -> list[Guardrail]:
        params = {"tag": tag} if tag else None
        return self._get_list("/guardrails", params=params, model=Guardrail)

    def get(self, guardrail_id: str) -> Guardrail:
        return self._get(f"/guardrails/{guardrail_id}", model=Guardrail)

    def update(self, guardrail_id: str, payload: GuardrailUpdate | Mapping[str, Any]) -> Guardrail:
        return self._put(f"/guardrails/{guardrail_id}", payload, model=Guardrail)

    def delete(self, guardrail_id: str) -> None:
        self._delete(f"/guardrails/{guardrail_id}")
